hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -161,24 +209,43 @@ private Connection getHBaseConnection() {
       }
       return taggedRecords.iterator();
     }
+
+    private Get generateStatement(String key) throws IOException {
+      return new Get(Bytes.toBytes(key)).setMaxVersions(1)
+          .addColumn(SYSTEM_COLUMN_FAMILY, COMMIT_TS_COLUMN)
+          .addColumn(SYSTEM_COLUMN_FAMILY, FILE_NAME_COLUMN)
+          .addColumn(SYSTEM_COLUMN_FAMILY, PARTITION_PATH_COLUMN);
+    }
+
+    private boolean checkIfValidCommit(String commitTs) {
+      HoodieTimeline commitTimeline = hoodieTable.getCompletedCommitTimeline();
```

Comment:
check if this will keep loading the timeline (which I think it does..).. You can make the commitTimeline a member of this class during iniitialization..

Reply:
It should'nt : https://github.com/uber/hudi/blob/master/hoodie-common/src/main/java/com/uber/hoodie/common/table/HoodieTableMetaClient.java#L173

I think you need to call reload() for that.

Reply:
ah there is some caching within table meta client. ok got it.